### PR TITLE
Make stop command blocking and return properly

### PR DESCRIPTION
**Pull request by mavam**
The command previously returned regardless of whether it succeeded.

### CODE REVIEW

File: libvast/src/system/stop_command.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+namespace vast::system {
+
+caf::message
+stop_command(const command::invocation& invocation, caf::actor_system& sys) {
+  VAST_TRACE(invocation);
+  // Bail out early for bogus invocations.
+  if (caf::get_or<bool>(invocation.options, "system.node", false))
+    return caf::make_message(
+      make_error(ec::parse_error, "cannot stop an immediately started node"));
```
Comment (dominiklohmann) [inline libvast/src/system/stop_command.cpp:38]:
```suggestion
      make_error(ec::parse_error, "cannot stop and immediately started node"));
```

File: libvast/src/system/stop_command.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+using namespace std::chrono_literals;
+
+namespace vast::system {
+
+caf::message
+stop_command(const command::invocation& invocation, caf::actor_system& sys) {
+  VAST_TRACE(invocation);
+  // Bail out early for bogus invocations.
+  if (caf::get_or<bool>(invocation.options, "system.node", false))
```
Comment (dominiklohmann) [inline libvast/src/system/stop_command.cpp:36]:
```suggestion
  if (caf::get_or(invocation.options, "system.node", false))
```

The template parameter can be derived from the fallback value.

File: libvast/src/system/stop_command.cpp [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+#include "vast/system/connect_to_node.hpp"
+
+#include <caf/actor.hpp>
+#include <caf/scoped_actor.hpp>
+#include <caf/settings.hpp>
+
+#include <iostream>
+
+using namespace std::chrono_literals;
+
```
Comment (dominiklohmann) [inline libvast/src/system/stop_command.cpp:29]:
```suggestion
```

These do not seem to be used anywhere.

**Comment by mavam** [comment]
Good findings, I've merged and squashed your suggestions.

File: libvast/src/system/start_command.cpp [hunk 4]
```diff
@@ -43,6 +43,10 @@ caf::message start_command_impl(start_command_extra_steps extra_steps,
                                 const command::invocation& invocation,
                                 caf::actor_system& sys) {
   VAST_TRACE(invocation);
+  // Bail out early for bogus invocations.
+  if (caf::get_or<bool>(invocation.options, "system.node", false))
```
Comment (dominiklohmann) [inline libvast/src/system/start_command.cpp:47]:
```suggestion
  if (caf::get_or(invocation.options, "system.node", false))
```

File: libvast/src/system/start_command.cpp [hunk 5]
```diff
@@ -43,6 +43,10 @@ caf::message start_command_impl(start_command_extra_steps extra_steps,
...
+    return caf::make_message(make_error(ec::parse_error, "cannot start a local "
+                                                         "node"));
```
Comment (dominiklohmann) [inline libvast/src/system/start_command.cpp:49]:
```suggestion
    return caf::make_message(make_error(ec::invalid_configuration, "cannot start a local node"));
```

This needs text reflow as well.

File: libvast/src/system/stop_command.cpp [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  // Obtain VAST node.
+  caf::scoped_actor self{sys};
+  auto node = connect_to_node(self, content(sys.config()));
+  if (!node)
+    return caf::make_message(std::move(node.error()));
+  self->monitor(*node);
+  // Delegate shutdown invocation to node.
+  caf::error err;
+  self->send_exit(*node, caf::exit_reason::user_shutdown);
+  self->receive(
```
Comment (dominiklohmann) [inline libvast/src/system/stop_command.cpp:44]: We should have an INFO message between the `send_exit` and `receive` calls, informing the user that a shutdown has been requested.

File: libvast/src/system/stop_command.cpp [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+namespace vast::system {
+
+caf::message
+stop_command(const command::invocation& invocation, caf::actor_system& sys) {
+  VAST_TRACE(invocation);
+  // Bail out early for bogus invocations.
+  if (caf::get_or(invocation.options, "system.node", false))
+    return caf::make_message(
+      make_error(ec::parse_error, "cannot stop and immediately started node"));
```
Comment (dominiklohmann) [inline libvast/src/system/stop_command.cpp:34]:
```suggestion
      make_error(ec::parse_error, "cannot stop and immediately start node"));
```

File: libvast/src/system/start_command.cpp (hunk 5 above)
Comment (mavam) [reply libvast/src/system/start_command.cpp:49]: This was `clang-format`. I cannot commit this if I apply this change.

Comment (dominiklohmann) [reply libvast/src/system/start_command.cpp:49]: I meant that you needed to do this change locally and then reformat for the text reflow. I've changed the error category.

File: libvast/src/system/stop_command.cpp (hunk 6 above)
Comment (mavam) [reply libvast/src/system/stop_command.cpp:44]: I've streamlined the logging in slightly different way, but this is now available.

**Comment by mavam** [comment]
Except for the `clang-format` issue, I've addressed all of your feedback.

**Pull request merged**